### PR TITLE
ci(label): use fork-action instead of action script

### LINKED ISSUE #3661: Use `fork-action` instead of the local one to handle label trigged checks
### Problem to Solve

switch to https://github.com/gear-tech/fork-action when we need to update the current implementation https://github.com/gear-tech/gear/blob/master/.github/actions/label/build.js again

### Possible Solution

Wrote https://github.com/gear-tech/fork-action last days with more static checks with javascript transpiler last days

### Notes

_No response_

### LINKED ISSUE #3821: Match the required checks in the labeling CI
### Problem to Solve

the labeling CI is broken after https://github.com/gear-tech/gear/pull/3631

### Possible Solution

include this fix in https://github.com/gear-tech/gear/pull/3764

### Notes

_No response_

### PR DESCRIPTION
**Pull request by clearloop**
Resolves #3661 
Resolves #3821 
Blocked by #3767

Migrates the action script of labeling to [gear-tech/fork-action](https://github.com/gear-tech/fork-action) for saving the time of debugging, make the action a github repo can have these benefits

- the action could be written in typescript
- it can have its own CI for testing but not make our current repo more verbose and bigger
- easier for version management

@gear-tech/dev 

---

Test: [32a63bb](https://github.com/gear-tech/gear/pull/3764/commits/32a63bb52981ea6b2c156aa70d523263e19f7317)

### CODE REVIEW

File: .github/workflows/label.yml [hunk 1]
```diff
@@ -13,19 +13,31 @@ jobs:
       || github.event.label.name == 'E2-forcemacos'
       || github.event.label.name == 'A4-insubstantial'
       || github.event.label.name == 'A2-mergeoncegreen'
+
     steps:
       - uses: actions/checkout@v4
+
+      - uses: gear-tech/fork-action@v0.1.0
+        if: github.event.label.name != 'E2-forcemacos'
```
Comment (breathx) [inline .github/workflows/label.yml:21]: This approach doesnt allow to run linux and macOS at the same time? :/

Comment (clearloop) [reply .github/workflows/label.yml]: https://github.com/gear-tech/gear/pull/3764/commits/b4ee445c9c43de3622bc4b207ff144fb86ef0001

Comment (clearloop) [reply .github/workflows/label.yml]: The labels here are triggered `async`, when we adding labels `A0-pleasereview` and `E2-forcemacos`, the dispatch workflow will be triggered for twice, once for `linux (A0)`, once for `macos (E2)`, see fc35380

**Review by breathx — APPROVED** [review]

**Comment by breathx** [comment]
@ark0f ptal

**Comment by breathx** [comment]
@clearloop just for clarification, does this approach work correctly with our release-debug jobs?
I'd tests it inside this PR with different labels

**Comment by clearloop** [comment]
> @clearloop just for clarification, does this approach work correctly with our release-debug jobs? I'd tests it inside this PR with different labels

It now only works for build jobs as what we have in our current code base

| E2-forcemacos | A0-pleasereview / A2-mergeoncegreen / A4-insubstantial |
|-|-|
| PR / build / macos-x86 | PR / build / linux (debug) |
| PR / build / macos-aarch64 | PR / build / win-cross (debug) |


Tag `E3-forcerelease` is currently not supported, but it could be added easily with this arch, btw `build / win-cross` could be outdated since it is not a required job anymore

for the re-testing, see db27d70

**Comment by clearloop** [comment]
> Need to support everything we could use on actual master before merging, including forcerelease label
> 
> Speaking of win-cross its okay not to be required, but it still runs each time, isn't it?

ideally, this PR provides the refactor of the `triggering checks on labelling` feature, without any modifications in results but the inside logic

`win-cross` is previous a `required check` ( if not pass, the PR could not be merged ) in our repo checks, it's no more a required check but still runs on merging into master ( and yes, this follows the ideal that supporting all checks running on merging ( master)  )

---

following the idea `running all checks using in master`, I can add supports for `forcenatwin`, `forcerelease` and `forceproduction` in this PR as well anyway

**Review by breathx — CHANGES_REQUESTED** [review]
waiting for force release

**Review by breathx — APPROVED** [review]
@ark0f take a look pls

good to go from my side

**Review by ark0f — APPROVED** [review]

**Comment by clearloop** [comment]
recheck everything at 0f3e90b

**Pull request merged**